src/spikeinterface/core/datasets.py

```diff
@@ -14,56 +14,80 @@ def download_dataset(
     remote_path: str = "mearec/mearec_test_10s.h5",
     local_folder: Path | None = None,
     update_if_exists: bool = False,
-    unlock: bool = False,
 ) -> Path:
     """
-    Function to download dataset from a remote repository using datalad.
+    Function to download dataset from a remote repository using a combination of datalad and pooch.
+
+    Pooch is designed to download single files from a remote repository.
+    Because our datasets in gin sometimes point just to a folder, we still use datalad to download
+    a list of all the files in the folder and then use pooch to download them one by one.
 
     Parameters
     ----------
     repo : str, default: "https://gin.g-node.org/NeuralEnsemble/ephy_testing_data"
         The repository to download the dataset from
     remote_path : str, default: "mearec/mearec_test_10s.h5"
         A specific subdirectory in the repository to download (e.g. Mearec, SpikeGLX, etc)
-    local_folder : str, default: None
+    local_folder : str, optional
```

Comment:
we use `default: None` elsewhere ;)

Reply:
Thanks @alejoe91!! Though please ignore  most of this PR ATM, it contains a lot of stray commits from random other PRs that are screwing everything up and I that need to drop! 😅

Reply:
Ok! I'll refrain myself ;)